packages/telemetry/src/make-slog-sender.js

```diff
@@ -1,32 +1,121 @@
 // @ts-check
 import path from 'path';
 import tmp from 'tmp';
+import { serializeSlogObj } from './serialize-slog-obj.js';
 
-export const DEFAULT_SLOG_SENDER_MODULE =
+export const DEFAULT_SLOGSENDER_MODULE =
   '@agoric/telemetry/src/flight-recorder.js';
+export const SLOGFILE_SENDER_MODULE = '@agoric/telemetry/src/slog-file.js';
 
-export const makeSlogSenderFromModule = async (
-  slogSenderModule = DEFAULT_SLOG_SENDER_MODULE,
-  makerOpts = {},
-) => {
-  if (!slogSenderModule) {
-    return undefined;
+/** @typedef {import('./index.js').SlogSender} SlogSender */
+
+/**
+ *
+ * @param {import('./index.js').MakeSlogSenderOptions} opts
+ */
+export const makeSlogSender = async (opts = {}) => {
+  const { env = {}, stateDir: stateDirOption, ...otherOpts } = opts;
+  const { SLOGSENDER = DEFAULT_SLOGSENDER_MODULE, ...otherEnv } = env;
+
+  const slogSenderModules = SLOGSENDER.split(',').map(modulePath =>
+    modulePath.startsWith('.')
+      ? // Resolve relative to the current working directory.
+        path.resolve(modulePath)
+      : modulePath,
+  );
+  if (
+    otherEnv.SLOGFILE &&
+    !slogSenderModules.includes(SLOGFILE_SENDER_MODULE)
+  ) {
+    slogSenderModules.push(SLOGFILE_SENDER_MODULE);
   }
 
-  if (slogSenderModule.startsWith('.')) {
-    // Resolve relative to the current working directory.
-    slogSenderModule = path.resolve(slogSenderModule);
+  if (!slogSenderModules.length) {
+    return undefined;
   }
 
-  console.warn(`Loading makeSlogSender from ${slogSenderModule}`);
-  const { makeSlogSender: maker } = await import(slogSenderModule);
-  if (typeof maker !== 'function') {
-    throw Error(`${slogSenderModule} did not export a makeSlogSender function`);
+  console.warn(
+    `Loading slog sender${
+      slogSenderModules.length > 1 ? 's' : ''
+    } from ${slogSenderModules.join(', ')}`,
+  );
```

Comment:
Please don't do math to construct natural language sentences.  It makes i18n more difficult, as well as being harder to parse mechanically.
```js
  console.warn('Loading slog sender modules:', slogSenderModules.join(', '));
```